syntax = "proto3";

// These Shares are unique to this device.

// Test Request and Publish
message Share1 {
    sfixed32 startingNumber = 1;
    sfixed32 endingNumber = 2;
    sfixed32 counter = 3; // Counts up each second from the starting number to the ending number
}

// Test device wiring configuration
message Share2 {
    sfixed32 frequencyInputPinId = 1;
    sfixed32 digitalOutputPinId = 2;
    sfixed32 analogInputAPinId = 3;
    sfixed32 analogInputBPinId = 4;
}

// Test device meta data
message Share3 {
    sfixed32 loopsPerSecond = 1;
}

// Test device String
message Share4 {
    string welcomeText = 1;
}

// Test device general types
message Share5 {
    float floatNumber = 1;
    double doubleNumber = 2;
    string ipAddress = 3;
    sfixed32 portNumber = 4;
    string datetimeUtc = 5; // ISO 8601 format
    bool booleanValue = 6;
}